waiter/src/waiter/service_description.clj

```diff
@@ -833,10 +835,10 @@
 
 (defn service-id->service-description
   "Loads the service description for the specified service-id including any overrides."
-  [kv-store service-id service-description-defaults metric-group-mappings &
-   {:keys [effective? nil-on-missing? refresh] :or {effective? true nil-on-missing? true refresh false}}]
-  (cond-> (fetch-core kv-store service-id :nil-on-missing? nil-on-missing? :refresh refresh)
-          effective? (default-and-override metric-group-mappings kv-store service-description-defaults service-id)))
+  [kv-store service-id service-description-defaults metric-group-mappings & {:keys [effective?] :or {effective? true}}]
+  (cond-> (or (fetch-core kv-store service-id :refresh false)
```

Comment:
Is there ever a time when we would want to continue with a `nil` value from `fetch-core` where we wouldn't want to perform refresh?  If so, does the double-checking belong in `fetch-core`?  

As I understand it, `refresh` forces a ZK `sync()`, which we'd always want to perform if we got a `nil` value.  Adding it to `fetch-core` ensures that any user of `fetch-core` benefits.

This would also make it unnecessary to make the above changes to the `suspend-or-resume-service-handler` and `override-service-handler` handlers.

Reply:
We discussed offline and decide to handle it by:
1. introducing the refresh behavior in the router state maintainer to prevent other subsystems from receiving the nil service descriptions
2. Maintaining the checks on the handlers which can receive the service-id externally.